lib/puppet/functions/stdlib/crc32.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+require 'zlib'
+# @note
+#   The CRC32 algorithm can easily generate collisions,
+#   but may be useful for generating sharding, describing
+#   secrets, or seeding nonce values.
+#
+# @summary
+#   Run a CRC32 calculation against a given value.
+Puppet::Functions.create_function(:'stdlib::crc32') do
+  # @param my_data The ScalarData to evaluate
+  # @example
```

Comment:
It's best to name your examples. You can see there's an empty header in the reference. Ideally speaking something would at least raise a warning, possibly an error.

Reply:
Added and squashed.